jumanji/environments/routing/sokoban/generator.py

```diff
@@ -0,0 +1,440 @@
+
+        Args:
+            key: the Jax random number generation key.
+
+        Returns:
+            state: the generated problem instance.
+        """
+
+    def get_agent_coordinates(self, grid: chex.Array) -> chex.Array:
+        """Extracts the coordinates of the agent from a given grid with the
+        assumption there is only one agent in the grid.
+
+        Args:
+            grid: Array (uint8) of shape (num_rows, num_cols)
+
+        Returns:
+            location: (int32) of shape (2,)
+        """
+
+        coordinates = jnp.where(grid == AGENT, size=1)
+
+        x_coord = jnp.squeeze(coordinates[0])
+        y_coord = jnp.squeeze(coordinates[1])
+
+        return jnp.array([x_coord, y_coord])
+
+
+class DeepMindGenerator(Generator):
+    """... (rest of the docstring remains unchanged)"""
+
+    def __init__(
+        self,
+        difficulty: str,
+        split: str,
+        proportion_of_files: float = 1.0,
+        verbose: bool = False,
+    ) -> None:
+        self.difficulty = difficulty
+        self.verbose = verbose
+        self.proportion_of_files = proportion_of_files
+
+        # Set the cache path to user's home directory's .cache sub-directory
+        self.cache_path = os.path.join(
+            os.path.expanduser("~"), ".cache", "sokoban_dataset"
+        )
+
+        # Downloads data if not already downloaded
+        self._download_data()
+
+        self.train_data_dir = os.path.join(
+            self.cache_path, "boxoban-levels-master", self.difficulty
+        )
+
+        if self.difficulty in ["unfiltered", "medium"]:
+            if self.difficulty == "medium" and split == "test":
+                raise Exception(
+                    "not a valid Deepmind Boxoban difficulty split" "combination"
+                )
+            self.train_data_dir = os.path.join(
+                self.train_data_dir,
+                split,
+            )
+
+        # Generates the dataset of sokoban levels
+        self._fixed_grids, self._variable_grids = self._generate_dataset()
+
+    def __call__(self, rng_key: chex.PRNGKey) -> State:
+        """Generate a random Boxoban problem from the Deepmind dataset.
+
+        Args:
+            rng_key: the Jax random number generation key.
+
+        Returns:
+            fixed_grid: Array (uint8) shape (num_rows, num_cols) the fixed
+            components of the problem.
+            variable_grid: Array (uint8) shape (num_rows, num_cols) the
+            variable components of the problem.
+        """
+
+        key, idx_key = jax.random.split(rng_key)
+        idx = jax.random.randint(
+            idx_key, shape=(), minval=0, maxval=self._fixed_grids.shape[0]
+        )
+        fixed_grid = self._fixed_grids.take(idx, axis=0)
+        variable_grid = self._variable_grids.take(idx, axis=0)
+
+        initial_agent_location = self.get_agent_coordinates(variable_grid)
+
+        state = State(
+            key=key,  # what key do we want to use for this
```

Comment:
This key is fine :smile: 

```suggestion
            key=key,
```